### PR TITLE
Fixed error response in password authenticator.

### PR DESCRIPTION
**Pull request by michsindelar**
Related to PR  #1343

### CODE REVIEW

File: addon/authenticators/oauth2-password-grant.js [hunk 1]
```diff
@@ -322,17 +322,24 @@ export default BaseAuthenticator.extend({
...
+      fetch(url, options)
+        .then((response) => {
+          response.text().then((text) => {
+            let json;
+            try {
+              json = JSON.parse(text);
+            } catch (e) {
+              json = {};
+            }
+            if (response.status === 200) {
```
Comment (marcoow) [inline addon/authenticators/oauth2-password-grant.js]: Can we continue using `response.ok` here?

Comment (michsindelar) [reply addon/authenticators/oauth2-password-grant.js]: If I'm right, response.ok does mean response.status >= 200 && response.status < 300 according to fetch, but RFC6749 allows only 200 for successful response.

Comment (marcoow) [reply addon/authenticators/oauth2-password-grant.js]: Yeah, relying on 200 would be a breaking change though and we'd need to include a deprecation if we see any other status codes. I'm all for doing that but should be a different PR.

Comment (michsindelar) [reply addon/authenticators/oauth2-password-grant.js]: It depends on you, I think this solution is fixing current bug. All responses, which are not in accordance with RFC6749 will be rejected.
But if you want, ill give there back response.ok;)

Comment (marcoow) [reply addon/authenticators/oauth2-password-grant.js]: Yes, please use `response.ok` for now - happy to merge another PR that adds a deprecation for status codes other than 200!

**Comment by michsindelar** [comment]
Sorry, Im not familiar with Ember tests.

**Comment by marcoow** [comment]
replaced by #1442

**Pull request closed without merging**